url354ce1a2d1_dagaren_gladiator_chess_tgz-pJ8-es_dagaren_gladiator_main_MainJ8/src/es/dagaren/gladiator/evaluation/Evaluator.java

```diff
@@ -0,0 +1,223 @@
+         -10, -20, -20, -20, -20, -20, -20, -10,
+         -20, -30, -30, -40, -40, -30, -30, -20,
+         -30, -40, -40, -50, -50, -40, -40, -30,
+         -30, -40, -40, -50, -50, -40, -40, -30,
+         -30, -40, -40, -50, -50, -40, -40, -30,
+         -30, -40, -40, -50, -50, -40, -40, -30,
+         
+                                            };
+   
+   public static final int[] endKingPieceSquare = {  
+         -50, -30, -30, -30, -30, -30, -30, -50,
+         -30, -30,   0,   0,   0,   0, -30, -30,
+         -30, -10,  20,  30,  30,  20, -10, -30,
+         -30, -10,  30,  40,  40,  30, -10, -30,
+         -30, -10,  30,  40,  40,  30, -10, -30,
+         -30, -10,  20,  30,  30,  20, -10, -30,
+         -30, -20, -10,   0,   0, -10, -20, -30,
+         -50, -40, -30, -20, -20, -30, -40, -50
+                                            };
+   
+   public static int[][] pieceSquareValue;
+   
+   static
+   {
+      pieceSquareValue = new int[6][];
+      
+      pieceSquareValue[GenericPiece.PAWN.index]   = pawnPieceSquare;
+      pieceSquareValue[GenericPiece.BISHOP.index] = bishopPieceSquare;
+      pieceSquareValue[GenericPiece.KNIGHT.index] = knightPieceSquare;
+      pieceSquareValue[GenericPiece.ROOK.index]   = rookPieceSquare;
+      pieceSquareValue[GenericPiece.QUEEN.index]  = queenPieceSquare;
+      pieceSquareValue[GenericPiece.KING.index]   = kingPieceSquare; 
+   }
+   
+   public int evaluate(Position position)
+   {
+      Colour turn = position.getTurn();
+      int score = 0;
+      
+      score += evaluateMaterial(position);
+      score += 0.2 * evaluateMobility(position);
+      score += evaluatePieceSquareTables(position);
+      
+      score += random.nextInt(10);
+      
+      if(turn == Colour.WHITE)
+         return score;
+      else
+         return -score;
+   }
+   
+   private int evaluateMaterial(Position position)
+   {
+      int score = 0;
+      
+      score += (position.getNumPieces(Piece.WHITE_PAWN)   - position.getNumPieces(Piece.BLACK_PAWN))   * PAWN_SCORE;
+      score += (position.getNumPieces(Piece.WHITE_ROOK)   - position.getNumPieces(Piece.BLACK_ROOK))   * ROOK_SCORE;
+      score += (position.getNumPieces(Piece.WHITE_BISHOP) - position.getNumPieces(Piece.BLACK_BISHOP)) * BISHOP_SCORE;
+      score += (position.getNumPieces(Piece.WHITE_KNIGHT) - position.getNumPieces(Piece.BLACK_KNIGHT)) * KNIGHT_SCORE;
+      score += (position.getNumPieces(Piece.WHITE_QUEEN)  - position.getNumPieces(Piece.BLACK_QUEEN))  * QUEEN_SCORE;
+      
+      
+      return score;
+   }
+   
+   private int evaluateMobility(Position position)
+   {
+      int whiteMobility = position.getMobility(Colour.WHITE);
+   
+      int blackMobility = position.getMobility(Colour.BLACK);
+      
+      return (whiteMobility == 0 && blackMobility == 0) ? 0 : ((whiteMobility - blackMobility) / (whiteMobility + blackMobility) * 100) ;
+   }
+   
+   private int evaluatePieceSquareTables(Position position)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 182-213 into a separate method.